src/cashier_frontend_new/src/modules/token/utils/converter.spec.ts

```diff
@@ -19,6 +19,22 @@ describe("balanceToIcp", () => {
   });
 });
 
+describe("icpToBalance", () => {
+  it("should throw an error for negative decimals", () => {
+    expect(() => icpToBalance(1, -1)).toThrow("Decimals cannot be negative");
+  });
+
+  it("should handle zero ICP", () => {
+    expect(icpToBalance(0, 8)).toBe(0n);
```

Comment:
As the converter is only used to show data in the UI, it should convert to a string instead of a number to avoid loss of precision in what is shown